### PR TITLE
Process LoRa Message only if comming from a device and can be deserialized

### PR DESCRIPTION
**Pull request by kbeaugrand**
## Description

What's new?

- Fix error in logs when trying to deserialize an event that is not coming from a LoRa device.

## What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Tests
- [ ] Other

### CODE REVIEW

File: src/AzureIoTHub.Portal.Server/Services/LoRaWanDeviceService.cs [hunk 1]
```diff
@@ -202,20 +220,18 @@ public override async Task ProcessTelemetryEvent(EventData eventMessage)
...
 
             loRaWanDevice.Telemetry
                 .OrderByDescending(telemetry => telemetry.EnqueuedTime)
-                .Skip(100)
+                .Skip(numberOfMessages)
                 .ToList()
                 .ForEach(telemetry =>
                 {
-                    this.deviceTelemetryRepository.Delete(telemetry.Id);
+                    _ = loRaWanDevice.Telemetry.Remove(telemetry);
```
Comment (hocinehacherouf) [inline src/AzureIoTHub.Portal.Server/Services/LoRaWanDeviceService.cs:233]: Telemetry entity is removed from lorawan device but not from database

**Review by hocinehacherouf — CHANGES_REQUESTED** [review]

File: src/AzureIoTHub.Portal.Server/Services/LoRaWanDeviceService.cs (hunk 1 above)
Comment (kbeaugrand) [reply src/AzureIoTHub.Portal.Server/Services/LoRaWanDeviceService.cs:233]: Maybe that's why you used SaveAsync twice during this action : 
- Removing from the collection
- removing from the table

You might have another option by setting non-nullable deviceId on the telemetry entity and specify that the context should delete the entity when removed from the collection ?

Comment (hocinehacherouf) [reply src/AzureIoTHub.Portal.Server/Services/LoRaWanDeviceService.cs:233]: We can move the method KeepOnlyLatestTelemetry to LoRaDeviceTelemetryRepository and use the DbContext to update the state of each telemetry entity. Example (not tested 😄):

```csharp
public void KeepOnlyLatestTelemetry(LorawanDevice loRaWanDevice, , int numberOfMessages = 100)
{
    loRaWanDevice.Telemetry
        .OrderByDescending(telemetry => telemetry.EnqueuedTime)
        .Skip(numberOfMessages)
        .ToList()
        .ForEach(telemetry =>
        {
            this.context.Entry(telemetry).State = Microsoft.EntityFrameworkCore.EntityState.Deleted;
        });
}
```

What do you think?

Comment (kbeaugrand) [reply src/AzureIoTHub.Portal.Server/Services/LoRaWanDeviceService.cs:237]: Context is not available in the services. 
I re-added the repository call to delete method

**Review by hocinehacherouf — APPROVED** [review]

**Pull request merged**